parsec/data.c

```diff
@@ -559,3 +562,28 @@ parsec_data_destroy( parsec_data_t *data )
 #endif
     PARSEC_OBJ_RELEASE(data);
 }
+
+void
+parsec_data_discard( parsec_data_t *data )
+{
+
+    /* first release the reference the application held */
+    PARSEC_OBJ_RELEASE(data);
+
+    /* second, mark the host copy as discarded */
+    parsec_data_copy_t *cpu_copy = data->device_copies[0];
```

Comment:
I'd rather keep the flag on the data copy because it's really the data copy that is in a discarded state. It doesn't have a reference to the data_t anymore, it just hangs around on the data_t for later release. Once we support removing the host copy entirely we can get rid of that.